go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -281,7 +282,10 @@ func (t *TxThrottler) Throttle() (result bool) {
 	if result {
 		t.requestsThrottled.Add(1)
 	}
-	return result
+
+	// Throttle according to both what the throttle state says, and the criticality. Workloads with higher criticality
+	// are less likely to be throttled.
+	return result && rand.Intn(100) < 100-int(criticality)
```

Comment:
the metric `requestsThrottled` should not be added when the decision is not to throttle.

Reply:
Fixed.